packages/features/bookings/lib/handleNewBooking/originalRescheduledBookingUtils.ts

```diff
@@ -5,24 +5,16 @@ import { HttpError } from "@calcom/lib/http-error";
 import { BookingRepository } from "@calcom/lib/server/repository/booking";
 import { BookingStatus } from "@calcom/prisma/enums";
 
-export const validateOriginalRescheduledBooking = async (
-  originalRescheduledBooking: OriginalRescheduledBooking
-) => {
-  if (!originalRescheduledBooking) {
+export async function getOriginalRescheduledBooking(uid: string, seatsEventType?: boolean) {
+  const originalBooking = await BookingRepository.findOriginalRescheduledBooking(uid, seatsEventType);
+
+  if (!originalBooking) {
     throw new HttpError({ statusCode: 404, message: "Could not find original booking" });
   }
 
-  if (
-    originalRescheduledBooking.status === BookingStatus.CANCELLED &&
-    !originalRescheduledBooking.rescheduled
-  ) {
-    throw new HttpError({ statusCode: 403, message: ErrorCode.CancelledBookingsCannotBeRescheduled });
+  if (originalBooking.status === BookingStatus.CANCELLED && !originalBooking.rescheduled) {
+    throw new HttpError({ statusCode: 400, message: ErrorCode.CancelledBookingsCannotBeRescheduled });
   }
-};
-
-export async function getOriginalRescheduledBooking(uid: string, seatsEventType?: boolean) {
-  const originalBooking = await BookingRepository.findOriginalRescheduledBooking(uid, seatsEventType);
-  validateOriginalRescheduledBooking(originalBooking);
```

Comment:
Looks like the reason for unhandled rejection was that await wasn't used here.

Shouldn't we just add await here? or was there some other reason?

Reply:
yeah this was the reason but logic in validateOriginalRescheduledBooking is not that complex/big so i moved that function to `getOriginalRescheduledBooking`